packages/evm/src/precompiles/05-modexp.ts

```diff
@@ -150,6 +149,7 @@ export function precompile05(opts: PrecompileInput): ExecResult {
     }
   }
 
+  const maxSize = opts.common.isActivatedEIP(7823) ? BigInt(1024) : BigInt(2147483647) // @ethereumjs/util setLengthRight limitation
```

Comment:
I think to be complicit with the EIP (or rather, before the EIP) a check should only be done if EIP 7823 is active (not hardcoding a "high" maxSize limit, this could fail specifically crafted tests which would not work in practice due to the high gas limit required to call this precompile with such large input data (due to quadratic memory gas costs))

Reply:
Are you suggesting we drop the fallback max size? It's A limit of `setLengthLeft`.

Reply:
Never mind, this was originally there. I thougt this semi-arbitrary (it is not) limit got added :smile: :+1: